test/langtools/tools/javac/LocalFreeVarStaticInstantiate.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * @test /nodynamiccopyright/
+ * @bug 8322882
+ * @summary Disallow attempts to access a free variable proxy field from a static method
+ * @compile/fail/ref=LocalFreeVarStaticInstantiate.out -XDrawDiagnostics LocalFreeVarStaticInstantiate.java
+ */
+
+class LocalFreeVarStaticInstantiate {
+
+    static void foo(Object there) {
+        class Local {
+            {
+                there.hashCode();
+            }
+            static {
+                new Local();    // can't get there from here
+            }
+        }
+    }
```

Comment:
I wonder if the test case should be enhanced with something like:
```
    Runnable r = () -> {
        Object there = "";
        class Local {
            {
                there.hashCode();
            }
            static {
                new Local();    // can't get there from here
            }
        }
    };
```

It works currently, as lambdas get a fake method owner if needed, so it would be just to make sure it remains working even if the handling of the lambdas changes.

Reply:
I've beefed up the test. While doing so I realized that the generated diagnostic was suboptimal, so I've fixed that too.